hal/ubuntu/qemu.sh

```
@@ -94,27 +205,76 @@ EOF
+  QEMU_OPTS="$QEMU_OPTS -drive if=pflash,format=raw,unit=0,file=$OVMF_CODE,readonly=on"
+  QEMU_OPTS="$QEMU_OPTS -drive if=pflash,format=raw,unit=1,file=$OVMF_VARS_COPY"
+  QEMU_OPTS="$QEMU_OPTS -cpu host"
+  QEMU_OPTS="$QEMU_OPTS -machine q35"
+  QEMU_OPTS="$QEMU_OPTS -device virtio-net-pci,netdev=vmnic,romfile="
+fi
+
+# Execute QEMU (use eval to handle complex quoting)
+echo "Full QEMU command:"
+echo "$QEMU_CMD $QEMU_OPTS"
+echo ""
+$QEMU_CMD $QEMU_OPTS
```

Shellcheck (0.11.0)

[warning] Quotes/backslashes in this variable will not be respected.

(SC2090)